### PR TITLE
Main-to-v3 update and V3 CLI

### LINKED ISSUE #2811: CLI: replace hyperlane-deploy with CLI for PI experience on v3
Blocked by main being merged into v3, which Yorke is doing atm. This is intended to be done prior to any work on getting the CLI to work with the new registry format

- [ ] update the CLI to work with v3
- [ ] update docs

### PR DESCRIPTION
**Pull request by jmrossy**
### Description

- Merge latest main into v3, including notably the new CLI package.
- Update CLI code for V3

### Drive-by changes

Remove stale token package ref from infra package.json

### Related issues

Fixes #2811

### Testing

Manual and CI test updates

### CODE REVIEW

File: typescript/cli/src/deploy/warp.ts [hunk 1]
```diff
@@ -329,8 +329,10 @@ function writeWarpUiTokenConfig(
       'No base Hyperlane token address deployed and no foreign deployment specified',
     );
   }
+  const chain = multiProvider.getChainMetadata(origin);
+  if (chain.protocol !== ProtocolType.Ethereum) throw Error('Unsupported VM');
   const commonFields = {
-    chainId: multiProvider.getChainId(origin),
+    chainId: multiProvider.getDomainId(origin),
```
Comment (jmrossy) [inline typescript/cli/src/deploy/warp.ts:335]: I think this should stay as `getChainId`. You can use `getChainIdNumber` if the types aren't happy

**Comment by avious00** [comment]
addresses: https://github.com/hyperlane-xyz/issues/issues/530

File: typescript/cli/src/config/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            ...result[chain],
+            [hookRequirements]: { type: HookType.MERKLE_TREE },
+          };
+          break;
+        }
+        case 'protocol_fee': {
+          const owner = await input({
+            message: 'Enter owner address',
+          });
+          const ownerAddress = ethers.utils.getAddress(owner);
```
Comment (jmrossy) [inline typescript/cli/src/config/hooks.ts:173]: Let's use our existing address related utils for this (`utils/src/addresses`). I'll fix now.

Comment (jmrossy) [inline typescript/cli/src/config/hooks.ts]: Let's use our existing address related utils for this (`utils/src/addresses`). I'll fix now.

File: typescript/cli/src/config/hooks.ts [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+          if (sameAsOwner) {
+            beneficiary = ownerAddress;
+          } else {
+            beneficiary = await input({
+              message: 'Enter beneficiary address',
+            });
+          }
+          const beneficiaryAddress = ethers.utils.getAddress(beneficiary);
+          // TODO: input in gwei, wei, etc
+          const maxProtocolFee = ethers.utils.parseUnits(
```
Comment (jmrossy) [inline typescript/cli/src/config/hooks.ts]: Ditto about utils, fixing now.

File: typescript/cli/src/deploy/core.ts [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+  log(`Found configs for chains: ${selectedChains.join(', ')}`);
+  return configs;
+}
+
+async function runHookStep(
+  selectedChains: ChainName[],
+  hookConfigPath?: string,
+) {
+  const presetConfigChains = Object.keys(presetHookConfigs);
+  console.log(
```
Comment (jmrossy) [inline typescript/cli/src/deploy/core.ts]: Nit: Use `log` from logger instead of console.log directly

File: typescript/cli/src/deploy/core.ts [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+  if (!chains?.length) {
+    chains = await runMultiChainSelectionStep(
+      customChains,
+      'Select chains to which core contacts will be deployed',
+    );
+  }
+  const artifacts = await runArtifactStep(chains, artifactsPath);
+  const multisigConfig = await runIsmStep(chains, ismConfigPath);
+  await runHookStep(chains, hookConfigPath);
```
Comment (jmrossy) [inline typescript/cli/src/deploy/core.ts]: Disabling for now since the value is unused

**Review by yorhodes — APPROVED** [review]

**Pull request merged**